### PR TITLE
Add possibility to save downloaded FIT data to file (reworked)

### PR DESCRIPTION
**Pull request by embear**
_This is an update of #42_

For backup purpose and to upload body measurements to platforms other than Garmin and TrainerRoad I would like to store the FIT or JSON data in a file. Because the original --no-upload mixes log messages and FIT or JSON data on stdout it is not possible to just redirect the output to a file. This patch adds --output to the list of arguments to specify a base filename where the FIT or JSON data is stored.

Attention: This patch changes the previous behavior of --no-upload which now doesn't print the FIT or JSON data to stdout.

### CODE REVIEW

**Comment by longstone** [comment]
Ok for me, but because there are breaking changes I'd like the go from the repo owner.

From my point of view, this improves the overall quality of this tool. It makes the switches, output, and behavior more predictable to the user.

The failing GitHub action is some weird issue that doesn't seem to be related to the contents of this PR.

TL;DR @jaroslawhartman can you check please:
```Attention: This patch changes the previous behavior of --no-upload which now doesn't print the FIT or JSON data to stdout.```
My suggestion is to merge this PR.

**Comment by jaroslawhartman** [comment]
Hi,

I agree, this is a good improvement. Though, we lose ability to write to stdout however does not look like anybody could use it anyway ("Because the original --no-upload mixes log messages and FIT or JSON data on stdout it is not possible to just redirect the output to a file").

Let's proceed with this change (after fixing the conflicts).

Best regards,
Jarek

**Comment by longstone** [comment]
Great! I'll merge this one.
